project/main.py

```diff
@@ -54,11 +56,22 @@ def get_task_status(id):
 
 @app.get("/task/{id}/result")
 def get_schedule_result(id):
-    # return StreamingResponse(iterfile(f'{id}_rostering.json'), media_type = "application/json")
+    if not task_exists(id):
```

Comment:
if not task_exists And result available

Reply:
task_exists just checks that dir exists. no dir == no results

Reply:
if dir exists - this doesn't mean, the results are available. should handle that gently